python-checks/src/main/java/org/sonar/python/checks/LineLengthCheck.java

```diff
@@ -39,44 +39,33 @@
     defaultValue = "" + DEFAULT_MAXIMUM_LINE_LENGTH)
   public int maximumLineLength = DEFAULT_MAXIMUM_LINE_LENGTH;
 
-  public int getMaximumLineLength() {
-    return maximumLineLength;
-  }
-
   @Override
-  public void visitFile(AstNode astNode) {
-    previousToken = null;
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.FILE_INPUT, ctx -> previousToken = null);
+    context.registerSyntaxNodeConsumer(Tree.Kind.TOKEN, ctx -> {
+      Token token = (Token) ctx.syntaxNode();
+      if (token.isGeneratedCode()) {
```

Comment:
I think we can drop that. I don't think that there's anything in Python which we consider as generated code. I would drop `org.sonar.python.api.tree.Token.isGeneratedCode()` as well.

Reply:
Newline tokens are actually considered generated code.

Dropping this altogether means we will raise issues on lines ending with a trailing comment, like: `foo = bar() # some very long comment` as the final \n will be used to determine the end of the line (instead of the closing parenthesis).

I assume we might actually want this to happen (the same way we might want to raise issues on lines only containing a # comment), but just to make clear that the behavior changes.

That said, I might simply rework this to ignore newline tokens and dropping the isGeneratedCode() method nevertheless.